test/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_string_buffer`
+
+class A {
+  String buffer;
+
+  void foo(int n) {
+    int aux = n;
+    while(aux --> 0) {
+      buffer += 'a'; // LINT
+    }
+  }
+}
```

Comment:
I think it would be better to catch a subset of cases and have zero false positives than to catch all possible cases and have some false positives. Studies (from other languages) have shown that lint rules that produce false positives are disabled, even when comments or other techniques can be used to silence the rule in specific locations, unless the rule is perceived as bringing a very significant value (which is *very* rare).

And when I say "false positives" I don't mean technical false positives, I mean lints produced in places where the user doesn't want a lint to have been produced, however technically accurate the lint might be.

That said, this is ultimately a judgement call. You know my opinion, now it's up to you and Alexei to make the call.